connector_ecommerce/models/stock.py

```diff
@@ -33,16 +33,23 @@ def do_transfer(self):
         self_context = self.with_context(__no_on_event_out_done=True)
         result = super(StockPicking, self_context).do_transfer()
         for picking in self:
-            if picking.picking_type_id.code != 'outgoing':
-                continue
-            if picking.related_backorder_ids:
-                method = 'partial'
-            else:
-                method = 'complete'
-            self._event('on_picking_out_done').notify(picking, method)
-            # deprecated:
-            on_picking_out_done.fire(self.env, 'stock.picking',
-                                     picking.id, method)
+            if picking.picking_type_id.code == 'outgoing':
```

Comment:
The following code can be done once, instead of twice.
It should better be written after the loop starts. 
if picking.related_backorder_ids:
+                    method = 'partial'
+                else:
+                    method = 'complete'

Reply:
one more little possibility(eliminate else):
method = 'complete'
if ...:
    method = 'partial'
OR
method = picking.related_backorder_ids and 'partial' or 'complete'.
I am not sure which is faster, but may be inline one.. @guewen

Reply:
The ternary form `a and b or c` should not be avoided because it can return something you don't expect when `b` has a false boolean value (not the case here but that's a good habit to adopt). Since python 2.5, there is `a if b else c`.
Performance-wise, the one-liner `if-else` is exactly the same.